duckduckgo/macos-browser
# Disable download reload on page tab reactivation/session restoration

Task/Issue URL: https://app.asana.com/0/1199237043630360/1201176142859594
Tech Design URL:
CC:

**Description**:
- Disables page reload when loading failed with error non-related to connectivity
- Disables reload of Tab started download on session restoration

**Steps to test this PR**:
1. Open Tab launching download (e.g. open an image https://app.asana.com/0/1196651294178074/1201123852249428/f and copy download link, open it in a new tab as it will auto-close if opened using the link)
2. After download has started activate another tab and reactivate the downloaded tab
3. Validate download isn't started again
4. Relaunch browser, validate download isn't started on the Tab restoration
5. Turn off wifi, validate navigation error is shown and Tab is tried to be reloaded on reactivation
6. Navigate to non-existent URL, validate Tab isn't reloaded on reactivation

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
